api/session_token_login_provider.go

```diff
@@ -73,11 +75,19 @@ func (p *sessionTokenLoginProvider) Login(ctx context.Context, caller base.APICa
 	return nil, errors.Trace(err)
 }
 
-func (p *sessionTokenLoginProvider) initiateDeviceLogin(ctx context.Context, caller base.APICaller) error {
-	if p.printOutputFunc == nil {
+func (p *sessionTokenLoginProvider) printOutput(format string, params ...any) error {
+	if p.cmdOut == nil {
 		return errors.New("cannot present login details")
 	}
```

Comment:
How can this be triggered? I'd rather prevent this in the construction of the provider.

Reply:
If a nil interface is passed to the constructor, one example is a non-interactive login trying to call the session token login provider.
I think the error needs to stay here. The `tryInOrderLoginProvider` is the default login provider which loops over a session token login provider and a usernamePassword login provider. That default login provider would fail during construction rather than correctly failing during execution and moving onto the next login provider.